File: src/main/java/libldt3/model/objekte/SendendesSystem.java
/*
 * Copyright 2016  Christoph Brill <egore911@gmail.com>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.objekte;

import libldt3.annotations.Feld;
import libldt3.annotations.Feldart;
import libldt3.annotations.Objekt;
import libldt3.annotations.Regelsatz;
import libldt3.model.regel.F007;
import libldt3.model.regel.F012;
import lombok.Getter;
import lombok.Setter;

/**
 * Dieses Objekt enthält die Information zum sendenden Softwaresystem, welches
 * den LDT Datensatz erstellt hat.
 */
@Objekt(value = "0051")
public @Getter @Setter class SendendesSystem {

	@Feld(value = "0001", feldart = Feldart.muss, regelsaetze = @Regelsatz(value = F007.class, maxLaenge = 12))
	private String version;
	@Feld(value = "8315", feldart = Feldart.kann, regelsaetze = @Regelsatz(maxLaenge = 60))
	private String empfaengerId;
	@Feld(value = "8316", feldart = Feldart.kann, regelsaetze = @Regelsatz(maxLaenge = 60))
	private String senderId;
	@Feld(value = "0105", feldart = Feldart.bedingt_muss, regelsaetze = @Regelsatz(value = F012.class, laenge = 16))
	private String kvbPruefnummer;
	@Feld(value = "8212", name = "Softwareverantwortlicher", feldart = Feldart.kann, regelsaetze = @Regelsatz(laenge = 24))
	private Organisation softwareverantwortlicher;
	@Feld(value = "0103", feldart = Feldart.muss, regelsaetze = @Regelsatz(maxLaenge = 60))
	private String softwareName;
	@Feld(value = "0132", feldart = Feldart.bedingt_muss, regelsaetze = @Regelsatz(maxLaenge = 60))
	private String softwareVersion;

}

File: cs/libldt3/model/objekte/SendendesSystem.cs
/*
 * Copyright 2016-2024  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using libldt3.attributes;
using libldt3.model;
using libldt3.model.regel.format;
using libldt3.model.regel.kontext;

namespace libldt3
{
    namespace model
    {
        namespace objekte
        {
            /// <summary>
            /// Dieses Objekt enthält die Information zum sendenden Softwaresystem, welches den
            /// LDT Datensatz erstellt hat.
            /// </summary>
            [Objekt(Value = "0051", Kontextregeln = new[] { typeof(K050) })]
            public class SendendesSystem : Kontext
            {
                [Objekt]
                public class SoftwareNameSoftware : Kontext
                {
                    public string Value;
                    [Feld(Value = "0132", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 60)]
                    public string VersionReleasestandSoftware;

                }
                [Feld(Value = "8315", Feldart = Feldart.kann)]
                [Regelsatz(MaxLaenge = 60)]
                public string DesEmpfaengersId;
                [Feld(Value = "8316", Feldart = Feldart.kann)]
                [Regelsatz(MaxLaenge = 60)]
                public string DesSendersId;
                [Feld(Value = "0105", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Value = new[] { typeof(F012) }, Laenge = 16)]
                public string KbvPruefnummer;
                [Feld(Value = "8212", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 24)]
                public Organisation Softwareverantwortlicher;
                [Feld(Value = "0103", Feldart = Feldart.muss)]
                [Regelsatz(MaxLaenge = 60)]
                public SoftwareNameSoftware SoftwareNameSoftware;

            }
        }
    }
}
